File: Front-End/src/app/models/workshop/tool.control/tool.control.material.ts
import { StatusEnabDisabEnum } from "../../status-enab-disab-enum";
import { TypeMaterialEnum } from "../type.material.enum";

export class ToolControlMaterial {
    companyId: number | null = null;
    resaleId: number | null = null;
    id: number | null = null;
    status: StatusEnabDisabEnum = StatusEnabDisabEnum.DISABLED;
    type: TypeMaterialEnum = TypeMaterialEnum.BOTH;
    numberCA: number | null = null;
    description: string = '';
    categoryId: number | null = null;
    quantityAccountingLoan: number = 0;
    quantityAvailableLoan: number = 0;
    quantityAccountingKit: number = 0;
    quantityAvailableKit: number = 0;
    validityDay: number | null = null;
    photoUrl: string = '';

    quantityLoan: number = 0;
    informationLoan: string = '';
}

File: API/src/main/java/com/concierge/apiconcierge/models/workshop/toolcontrol/ToolControlMaterial.java
package com.concierge.apiconcierge.models.workshop.toolcontrol;

import com.concierge.apiconcierge.models.enums.StatusEnableDisable;
import com.concierge.apiconcierge.models.workshop.toolcontrol.enums.TypeRequest;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
@Entity
@Table(name = "tb_tool_control_material")
public class ToolControlMaterial {

    @Column(name = "company_id")
    private Integer companyId;

    @Column(name = "resale_id")
    private Integer resaleId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private StatusEnableDisable status;

    private TypeRequest type;

    @Column(name = "number_ca")
    private Integer numberCA;

    private String description;

    @Column(name = "category_id")
    private Integer categoryId;

    @Column(name = "quantity_accounting_loan")
    private BigDecimal quantityAccountingLoan;

    @Column(name = "quantity_available_loan")
    private BigDecimal quantityAvailableLoan;

    @Column(name = "quantity_accounting_kit")
    private BigDecimal quantityAccountingKit;

    @Column(name = "quantity_available_kit")
    private BigDecimal quantityAvailableKit;

    @Column(name = "validity_day")
    private Integer validityDay;

    @Column(name = "photo_url")
    private String photoUrl;
}
